### PR TITLE
Refactor Input component

### PR DESCRIPTION
**Pull request by pawelacio**
- Refactor Input component
- Add tests

### CODE REVIEW

File: src/components/Input/Input.js [hunk 1]
```diff
@@ -14,83 +14,81 @@ import {
...
+  floatingLabels,
+  reference
+}) => {
+  return (
+    <InputComponentStyled>
+      <InputElementWrapperStyled
+        error={error}
+        passwordStrength={passwordStrength}
+      >
+        {/* {icon && <InputIconStyled>{icon.render()}</InputIconStyled>} */}
```
Comment (iwonakulacz) [inline src/components/Input/Input.js:41]: can this be deleted?

File: src/components/Input/Input.js [hunk 2]
```diff
@@ -14,83 +14,81 @@ import {
...
+        passwordStrength={passwordStrength}
+        id={`${placeholder}-desc`}
+      >
+        {error}
+      </ErrorWrapper>
+    </InputComponentStyled>
+  );
+};
 
 Input.propTypes = {
```
Comment (iwonakulacz) [inline src/components/Input/Input.js:93]: I would put a time on the props definition here. Be as precise as we can witch prop validation here, eg. type should oneOf with defined available values

File: src/components/Input/Input.js [hunk 3]
```diff
@@ -14,83 +14,81 @@ import {
...
+  showVisibilityIcon,
+  handleClickShowPassword,
+  showPassword,
+  passwordStrength,
+  ariaRequired,
+  ariaInvalid,
+  icon,
+  required,
+  floatingLabels,
+  reference
```
Comment (iwonakulacz) [inline src/components/Input/Input.js:33]: worth checking if we are using all of these props when embedding an Input component. Maybe we could remove unused ones

Comment (pawelacio) [reply src/components/Input/Input.js:33]: It looks that all of the props are used but maybe in the process of refactoring components like DateInput, EmailInput some will be deleted. 
image

File: src/components/Input/Input.js (hunk 1 above)
Comment (pawelacio) [reply src/components/Input/Input.js:41]: I was thinking about that line but you're right. I will delete it

**Review by iwonakulacz — APPROVED** [review]

**Pull request merged**